lib/vdsm/API.py

```diff
@@ -1809,11 +1809,14 @@ class ManagedVolume(APIBase):
 
     @api.logged(on="api.storage")
     @api.method
-    def attach_volume(self, vol_id, connection_info):
+    def attach_volume(self, vol_id, connection_info, sd_id=None):
         """
         attach volume and return attached device information
         """
-        return managedvolume.attach_volume(vol_id, connection_info)
+        return managedvolume.attach_volume(
+            vol_id,
+            connection_info,
+            sd_id=sd_id)
```

Comment:
The internal managedvolume.attach_volume() can be change to accept sd_id as first arguemnt, in the same way we pass it in other places, so this becomes:

    return managedvolume.attach_volume(sd_id, vol_id, connection_info)